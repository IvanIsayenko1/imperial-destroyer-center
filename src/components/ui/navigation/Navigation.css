nav {
  display: flex;
  gap: 24px;
  padding: 1rem 0;
}

.nav-link {
  color: var(--rebel-base);
  text-decoration: none;
  font-size: 1rem;
  font-weight: 500;
  padding: 0.75rem 1.5rem;
  border-radius: 8px;
  transition: all 0.3s ease;
  position: relative;
  background: rgba(0, 0, 0, 0.2);
  border: 1px solid rgba(255, 59, 59, 0.1);
  text-transform: uppercase;
  letter-spacing: 1px;
}

.nav-link::before {
  content: "";
  position: absolute;
  inset: 0;
  background: linear-gradient(45deg, var(--sith-red), transparent);
  opacity: 0;
  transition: opacity 0.3s ease;
  border-radius: 8px;
  z-index: -1;
}

.nav-link:hover {
  color: white;
  transform: translateY(-2px);
}

.nav-link:hover::before {
  opacity: 1;
}

.nav-link.active {
  color: white;
  background: linear-gradient(45deg, var(--sith-red), transparent);
  border-color: var(--sith-red);
  box-shadow: 0 0 20px rgba(255, 59, 59, 0.2);
}
