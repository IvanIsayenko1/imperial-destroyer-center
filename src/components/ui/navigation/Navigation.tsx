import React from "react";
import { NavLink } from "react-router-dom";
import "./Navigation.css";

interface NavigationItem {
  label: string;
  href: string;
}

interface NavigationProps {
  items?: NavigationItem[];
}

export const Navigation: React.FC<NavigationProps> = ({
  items = [
    { label: "Planets", href: "/planets" },
    { label: "Starships", href: "/starships" },
    { label: "Vehicles", href: "/vehicles" },
  ],
}) => {
  return (
    <nav>
      {items.map((item) => (
        <NavLink
          key={item.href}
          to={item.href}
          className={({ isActive }) => `nav-link ${isActive ? "active" : ""}`}
        >
          {item.label}
        </NavLink>
      ))}
    </nav>
  );
};
