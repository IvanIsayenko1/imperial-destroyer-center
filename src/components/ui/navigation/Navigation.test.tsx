import { render, screen } from "@testing-library/react";
import { Navigation } from "./Navigation";
import { MemoryRouter } from "react-router-dom";

describe("Navigation component", () => {
  it("should render navigation links", () => {
    const items = [
      { label: "Planets", href: "/planets" },
      { label: "Starships", href: "/starships" },
      { label: "Vehicles", href: "/vehicles" },
    ];

    render(
      <MemoryRouter initialEntries={["/planets"]}>
        <Navigation items={items} />
      </MemoryRouter>
    );

    items.forEach((item) => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    });
  });

  it("should apply the active class to the correct link", () => {
    const items = [
      { label: "Planets", href: "/planets" },
      { label: "Starships", href: "/starships" },
      { label: "Vehicles", href: "/vehicles" },
    ];

    render(
      <MemoryRouter initialEntries={["/starships"]}>
        <Navigation items={items} />
      </MemoryRouter>
    );

    // Check if "Starships" link is active
    const starshipsLink = screen.getByText("Starships");
    expect(starshipsLink).toHaveClass("active");

    // Check if other links are not active
    const planetsLink = screen.getByText("Planets");
    expect(planetsLink).not.toHaveClass("active");

    const vehiclesLink = screen.getByText("Vehicles");
    expect(vehiclesLink).not.toHaveClass("active");
  });

  it("should render default navigation items when no items prop is passed", () => {
    render(
      <MemoryRouter initialEntries={["/planets"]}>
        <Navigation />
      </MemoryRouter>
    );

    const defaultItems = [
      { label: "Planets", href: "/planets" },
      { label: "Starships", href: "/starships" },
      { label: "Vehicles", href: "/vehicles" },
    ];

    defaultItems.forEach((item) => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    });
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/planets"]}>
        <Navigation />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
