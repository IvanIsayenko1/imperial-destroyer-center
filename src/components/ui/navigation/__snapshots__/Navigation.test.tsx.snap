// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation component > should match snapshot 1`] = `
<DocumentFragment>
  <nav>
    <a
      aria-current="page"
      class="nav-link active"
      data-discover="true"
      href="/planets"
    >
      Planets
    </a>
    <a
      class="nav-link "
      data-discover="true"
      href="/starships"
    >
      Starships
    </a>
    <a
      class="nav-link "
      data-discover="true"
      href="/vehicles"
    >
      Vehicles
    </a>
  </nav>
</DocumentFragment>
`;
